src/mempool/storage.rs

```diff
@@ -494,30 +500,44 @@ pub trait Storage {
     }
 
     /// Creates and saves a new DataProposal if there are pending transactions
-    fn new_data_proposal(&mut self, crypto: &BlstCrypto, txs: Vec<Transaction>) -> Result<()> {
+    fn new_data_proposal(
+        &mut self,
+        crypto: &BlstCrypto,
+        txs: Vec<Transaction>,
+    ) -> Result<DataProposalCreation> {
```

Comment:
DataProposalCreation could be an Option instead?

Reply:
Result<Option<>> seems like a decently common pattern